InterGalacticCurrencyConvertor/src/main/java/materials/Materials.java

```diff
@@ -0,0 +1,57 @@
+package main.java.materials;
+
+import java.util.HashMap;
+import java.util.Objects;
+
+/**
+ * Keeps track of different types of materials that can be sold InterGalactically.
+ */
+public class Materials {
+	private HashMap<String, Material> mMap  = new HashMap<>();
+	private static Materials mInstance = new Materials();
+	public static Materials getInstance() {
+		return mInstance;
+	}
+	public void addMaterial(Material inMaterial) {
+	  Objects.requireNonNull(inMaterial, "Invalid Material Name");
+		mMap.put(inMaterial.getMaterialName(), inMaterial);
+	}
+	public boolean isMaterialAvailable(String inMaterialName) {
+		return mMap.containsKey(inMaterialName);
+	}
+	public double getMaterialUnitPrice(String inMaterialName) {
+	  Objects.requireNonNull(inMaterialName, "Invalid Material Name");
+		if (!isMaterialAvailable(inMaterialName)) {
+			throw new IllegalArgumentException("Invalid Material Name provide");
+		}
+		return mMap.get(inMaterialName).getMaterialUnitPrice();
+	}
+	public void createMaterial(String inMaterialName,
+	                           int inNumUnits,
+	                           double inTotalCost) {
+	  Objects.requireNonNull(inMaterialName, "Invalid Material Name");
+		Material material = new Material(inMaterialName, inNumUnits, inTotalCost);
+		Materials.getInstance().addMaterial(material);
+	}
+}
+
+/**
+ * Defines a material and its unit cost.
+ */
+class Material {
+	private String mName;
+	private double mUnitPrice;
+	public Material(String inMaterialName, int inNumUnits, double inTotalCost) {
+		if (inMaterialName == null) {
+			throw new NullPointerException("No Material Name provided");
+		}
+		mName = inMaterialName;
+		mUnitPrice = inTotalCost / inNumUnits;
+	}
+	public String getMaterialName() {
```

Comment:
Remove 'Material' from the method name here and below. It is redundant.

Reply:
updated